prefect.yaml

```
@@ -4,31 +4,64 @@
 
 # Generic metadata about this project
 name: haxpes-workflows
-prefect-version: 2.19.7
+prefect-version: 3.6.0
 
 # build section allows you to manage and build docker images
 build:
+  - prefect.deployments.steps.run_shell_script:
+      id: get-commit-hash
+      script: git rev-parse --short HEAD
+      stream_output: false
+  - prefect_docker.deployments.steps.build_docker_image:
+      id: build-image
+      requires: prefect-docker>0.6.0
+      image_name: ghcr.io/nsls2/haxpes-workflow-prefect3
+      tag: "{{ get-commit-hash.stdout }}"
+      dockerfile: Dockerfile
+      additional_tags:
+        - main
+
 
 # push section allows you to manage if and how this project is uploaded to remote locations
 push:
+  - prefect_docker.deployments.steps.push_docker_image:
+      requires: prefect-docker>0.6.0
+      image_name: "{{ build-image.image_name }}"
+      tag: "{{ build-image.tag }}"
+      credentials:
+        "{{
+        prefect.blocks.docker-registry-credentials.ghcr-nsls2-haxpes-workflow-prefect3
+        }}"
+      additional_tags:
+        - main
 
 # pull section allows you to provide instructions for cloning this project in remote locations
 pull:
+  - prefect.deployments.steps.set_working_directory:
+      directory: /repo
   - prefect.deployments.steps.git_clone:
-      repository: https://github.com/NSLS-II-SST/workflows-haxpes.git
-      branch: main
-      access_token:
+      repository: https://github.com/JunAishima/workflows-haxpes.git
+      branch: prefect3-pixi-upgrade
 
 # the deployments section allows you to provide configuration for deploying flows
 deployments:
-  - name: haxpes-end-of-run-workflow
-    version:
+  - name: haxpes-end-of-run-workflow-docker
+    version: 0.1.0
     tags: [haxpes]
```

Additional tag for branch to keep things consistent.
```suggestion
    tags:
      - haxpes
      - main
```